### ISSUE TITLE
Rustdoc and rustc accept different `cfg` syntax

### ISSUE THREAD

**Issue by jyn514** [rust-lang/rust#84437]
I tried this code:

```rust
#![feature(doc_cfg)]
#[doc(cfg(x, y))]
fn bar() {}
```

I expected to see this happen: An error, to match the behavior on `#[cfg(x, y)]`:

```
error: multiple `cfg` predicates are specified
 --> cfg2.rs:1:10
  |
1 | #[cfg(x, y)]
  |          ^
```

Instead, this happened: Rustdoc silently accepts the code.

Relevant code: https://github.com/rust-lang/rust/blob/88b99dec2a4d103b8153ca9300bf0fbebdf65bda/src/librustdoc/clean/cfg.rs#L62
I think the relevant rustc code is https://github.com/rust-lang/rust/blob/88b99dec2a4d103b8153ca9300bf0fbebdf65bda/compiler/rustc_expand/src/config.rs#L455

### Meta

`rustdoc --version`: rustdoc 1.53.0-nightly (392ba2ba1 2021-04-17)

**Comment by jyn514** [comment]
Fixing this properly will be hard. Rustc doesn't store the cfg predicates, it evaluates them immediately and returns whether they're active in the current crate. I guess an ok solution for now is to throw lots of tests at it? I'd be worried we're missing things, though.

@petrochenkov do you have suggestions for unifying the parsing somehow? For reference, this is what rustdoc currently uses: https://doc.rust-lang.org/nightly/nightly-rustc/rustdoc/clean/cfg/enum.Cfg.html
```rust
    True,
    False,
    Cfg(Symbol, Option<Symbol>),
    Not(Box<Cfg>),
    Any(Vec<Cfg>),
    All(Vec<Cfg>),
}
```
It's ok for the enum to be slightly different, but it can't just be a boolean true/false because rustdoc needs to display it.
